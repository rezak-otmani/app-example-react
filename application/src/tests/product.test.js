import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Product from '../views/product/product';

describe('Product Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Product />
    );
  });

           it('it should render Product components ', () => {
              expect(wrapper.length).toBe(1);

          });
          
           it('it should render the title', () => {
             const item = wrapper.find('h1').first();
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('OUR TECHNOLOGY')
           });


describe('Fast section', () => {

             it('it shows the Fast title', () => {
             const item = wrapper.find('.content').at(0);
             expect(item.length).toEqual(1);
             expect(item.childAt(1).text()).toBe(' FAST')
           });
             
            it('it shows the Fast paragraphe', () => {
             const item = wrapper.find('.content').at(0);
             expect(item.length).toEqual(1);
             expect(item.childAt(2).exists()).toBe(true)
           });

});


describe('Secure section', () => {

          it('it shows the Secure title ', () => {

             const item = wrapper.find('.content').at(1);
             expect(item.length).toEqual(1);
             expect(item.childAt(1).text()).toBe(' SECURE')
           });
           
             it('it shows the Secure paragraphe', () => {
             const item = wrapper.find('.content').at(1);
             expect(item.length).toEqual(1);
             expect(item.childAt(2).exists()).toBe(true)
           });

 });


describe('Easy section', () => {

          it('it shows the Easy title ', () => {

             const item = wrapper.find('.content').at(2);
             expect(item.length).toEqual(1);
             expect(item.childAt(1).text()).toBe(' EASY')
           });

             it('it shows the Easy paragraphe', () => {
             const item = wrapper.find('.content').at(2);
             expect(item.length).toEqual(1);
             expect(item.childAt(2).exists()).toBe(true)
           });

 });


describe('Image section', () => {

          it('it should render the screen image ', () => {

             const im = wrapper.find('.ui reveal').first();
             expect(im.length).toEqual(1);
             
             const img = wrapper.find('img');
             expect(img.length).toEqual(2);

           });

 });


describe('Button to try demo', () => {

          it('it should render the button to scroll in demo section ', () => {

             const link = wrapper.find({to: 'demo'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Try a demo')

           });

 });


          const tree = renderer.create(
                      <Product />
                          ).toJSON();
                  expect(tree).toMatchSnapshot();



});

