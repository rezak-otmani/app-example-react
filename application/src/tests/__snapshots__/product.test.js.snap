// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="pusher"
>
  <div
    className="ui vertical masthead center aligned segment"
  >
    <div
      className="center aligned column"
    >
      <div
        className="ui sizer vertical segment"
      >
        <h1
          className="ui huge header"
        >
          OUR TECHNOLOGY
        </h1>
      </div>
    </div>
    <div
      className="ui seven columns center aligned stackable grid"
    >
      <div
        className="column"
      >
        <h2
          className="ui left aligned header"
        >
          <div
            className="content"
          >
            <span
              className="ui olive header"
            >
               01 /
            </span>
             FAST
            <div
              className="sub header"
            >
              I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your                                                                   content and make changes to the font. Feel free to drag and drop me a ywhere you like on your page.
            </div>
          </div>
        </h2>
      </div>
      <div
        className="column"
      >
        <h2
          className="ui left aligned header"
        >
          <div
            className="content"
          >
            <span
              className="ui olive header"
            >
                02 /
            </span>
             SECURE
            <div
              className="sub header"
            >
              I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your own                                                                           content and make changes to the font. Feel free to drag and drop me an ywhere you like on your page.
            </div>
          </div>
        </h2>
      </div>
      <div
        className="column"
      >
        <h2
          className="ui left aligned header"
        >
          <div
            className="content"
          >
            <span
              className="ui olive header"
            >
                 03 /
            </span>
             EASY
            <div
              className="sub header"
            >
              I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your                                                                   own content and make changes to the font. Feel free to drag and drop me anywhere you like on your page.
            </div>
          </div>
        </h2>
      </div>
    </div>
    <div
      className="ui three columns center aligned grid"
    >
      <div
        className="ui reveal"
      >
        <div
          className="visible content"
        >
          <img
            alt=""
            className="ui Big image"
            src="graphics.png"
          />
        </div>
        <div
          className="hidden content"
        >
          <img
            alt=""
            className="move left"
            src="ecran.jpg"
          />
        </div>
      </div>
    </div>
    <a
      className="header item"
      href="#demo"
      onClick={[Function]}
    >
      <div
        className="positive massive ui button"
      >
        Try a demo
        <i
          className="right arrow icon"
        />
      </div>
    </a>
    <br />
    <br />
    <br />
  </div>
</div>
`;
